packages/@aws-cdk/assert/lib/test-stack.ts

```diff
@@ -0,0 +1,7 @@
+import cdk = require('@aws-cdk/cdk');
+
+export class TestStack extends cdk.Stack {
+  public testInvokeAspects() {
+    this.invokeAspects();
```

Comment:
Should we move  invokeAspects behind the all new “prepareTree”?

Reply:
Yes I think that makes sense.

Reply:
Just for clarity, I am doing:

1. remove `invokeAspects()` from `app.ts`
1. putting `invokeAspects()` inside of `prepareTree()`
1. removing the `TestStack` because `prepareTree()` is public and called during `expect`

Is there any concern about the conditional wrapped around the block that calls `prepareTree()`?

```ts
    if (!skipValidation) { // any concern here - I haven't found any but just double checking
      // Do a prepare-and-validate run over the given stack
      stack.node.prepareTree();

      const errors = stack.node.validateTree();
      if (errors.length > 0) {
        throw new Error(`Stack validation failed:\n${errors.map(e => `${e.message} at: ${e.source.node.scope}`).join('\n')}`);
      }
    }

// this will be removed
    if (isTestStack(stack)) {
      stack.testInvokeAspects();
    }
// end of block to be removed
```
A simple thumbs up to verify is fine.

Reply:
Lgtm; Curious why prepare tree is skipped if validation is skipped but that’s for a different issue (raise one?).